src/app/components/applications/SSV/MyAccount/components/Validator/BulkActions/BulkComponent.tsx

```diff
@@ -71,41 +79,47 @@ const BulkComponent = () => {
   };
 
   const nextStep = async () => {
+    const selectedValidatorKeys =  Object.keys(selectedValidators);
+    const selectedValidatorValues =  Object.values(selectedValidators);
     let res;
-    const condition = selectedValidators.length === 1;
+    const condition = selectedValidatorValues.every((validator: { validator: IValidator, isSelected: boolean }) => validator.isSelected);
     if (currentStep === BULK_STEPS.BULK_ACTIONS) {
       setCurrentStep(BULK_STEPS.BULK_CONFIRMATION);
     } else if (currentStep === BULK_STEPS.BULK_CONFIRMATION && currentBulkFlow === BULK_FLOWS.BULK_EXIT) {
-      res = await conditionalExecutor(condition, async () => await validatorStore.exitValidator(`0x${selectedValidators[0]}`, process.item.operators.map((operator: IOperator) => operator.id)), async () => await validatorStore.bulkExitValidators(selectedValidators.map((publicKey: string) => `0x${publicKey}`), process.item.operators.map((operator: IOperator) => operator.id)));
+      const singleFormattedPublicKey = formatValidatorPublicKey(selectedValidatorKeys[0]);
+      res = await conditionalExecutor(condition, async () => await validatorStore.exitValidator(singleFormattedPublicKey, process.item.operators.map((operator: IOperator) => operator.id)), async () => await validatorStore.bulkExitValidators(selectedValidatorKeys.filter((publicKey: string) => selectedValidators[publicKey].isSelected), process.item.operators.map((operator: IOperator) => operator.id)));
```

Comment:
I think the code will be a bit cleaner if we extract the functions const exitSIngleFunc = validatorStore.exitValidator and const exitBulkFunc = validatorStore.bulkExitValidators as variable and then call the function conditionalExecutor(condition, exitSingle, exitBulk))

const exitSingle = async () => await validatorStore.exitValidator(singleFormattedPublicKey, process.item.operators.map((operator: IOperator) => operator.id));
      const exitBulk = async () => await validatorStore.bulkExitValidators(selectedValidatorKeys.filter((publicKey: string) => selectedValidators[publicKey].isSelected), process.item.operators.map((operator: IOperator) => operator.id));
      res = await conditionalExecutor(condition, exitSingle, exitBulk);

Reply:
done